stock_valuation_fifo_lot/wizard/stock_valuation_layer_revaluation.py

```diff
@@ -0,0 +1,244 @@
+        for reval in revaluations:
+            reval.new_value = reval.current_value_lot_svl + reval.added_value
+            if not float_is_zero(
+                reval.current_quantity_lot_svl,
+                precision_rounding=self.product_id.uom_id.rounding,
+            ):
+                reval.new_value_by_qty = (
+                    reval.new_value / reval.current_quantity_lot_svl
+                )
+            else:
+                reval.new_value_by_qty = 0.0
+        return super(
+            StockValuationLayerRevaluation, self - revaluations
+        )._compute_new_value()
+
+    def action_validate_revaluation(self):
+        self.ensure_one()
+        if self.lot_id:
+            # Overrides standard behavior to store the remaining value in a new SVL rather than
+            #  modifying the existing one.
+            if self.currency_id.is_zero(self.added_value):
+                raise UserError(
+                    _("The added value doesn't have any impact on the stock valuation")
+                )
+            product_id = self.product_id.with_company(self.company_id)
+            remaining_svls = self.env["stock.valuation.layer"].search(
+                [
+                    ("product_id", "=", product_id.id),
+                    ("remaining_qty", ">", 0),
+                    ("company_id", "=", self.company_id.id),
+                    ("lot_ids", "in", self.lot_id.id),
+                ]
+            )
+            remaining_svls = remaining_svls.filtered(
+                lambda l: l._get_unconsumed_in_move_line(self.lot_id)
+            )
+            if self.reason:
+                description = _("Manual Stock Valuation: %s.", self.reason)
+            else:
+                description = _("Manual Stock Valuation: No Reason Given.")
+            if product_id.categ_id.property_cost_method == "average":
+                description += _(
+                    " Product cost updated from %(previous)s to %(new_cost)s.",
+                    previous=product_id.standard_price,
+                    new_cost=product_id.standard_price
+                    + self.added_value / self.current_quantity_lot_svl,
+                )
+            revaluation_svl_vals = {
+                "company_id": self.company_id.id,
+                "product_id": product_id.id,
+                "description": description,
+                "value": self.added_value,
+                "lot_ids": [Command.set([self.lot_id.id])],
+                "is_lot_revaluation": True,
+                "remaining_value": self.added_value,
+                "quantity": 0,
+            }
+            remaining_qty = sum(remaining_svls.mapped("remaining_qty"))
+            remaining_value = self.added_value
+            remaining_value_unit_cost = self.currency_id.round(
+                remaining_value / remaining_qty
+            )
+            svls = self.env["stock.valuation.layer"]
+            for svl in remaining_svls:
```

Comment:
Just to improve the readability.
```suggestion
            rounding = product.uom_id.rounding
            for svl in remaining_svls:
```